### ISSUE TITLE
Bundled AMD dependency resolution

### ISSUE THREAD

**Issue by khangiskhan** [systemjs/systemjs#1980]
When using SystemJS (latest version 5.0.0) with AMD modules, if a module has a dependency that has not been loaded in the registry yet, it will make a request for it.  If the requested module is inside of an AMD bundle, it is not "resolved" until it is requested again via `import`. (This is not a problem if the dependency is already in the registry prior to the request).

This is similar to issue https://github.com/systemjs/systemjs/issues/1037, and https://stackoverflow.com/questions/42520815/loading-bundled-amd-modules-with-systemjs, the latter of which has a solution, but is for an older version of SystemJS.

```
// importmap.json
{
  imports: {
   'app': '/js/app.js',
   'test': '/js/test.js'
  }
}
```

```
// test.js
define('testDep', [], function(){});
define('test', ['testDep'] function(testDep) {
  var test = 'hello world';
  return test;
});

```

```
// app.js
define('app', ['test'], function(test) {
  console.log(test); // empty object
  // workaround impractical hack:
  System.import('test').then( t => console.log(t)); // { default: 'hello world, useDefault: true }
});
```

```
<!-- index.html -->
<html>
  <head>
    <script type="systemjs-importmap" src="importmap.json">
    </script>
    <script src="https://cdn.jsdelivr.net/npm/systemjs@5.0.0/dist/system.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/systemjs@5.0.0/dist/extras/amd.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/systemjs@5.0.0/dist/extras/named-register.js">
</script>
    <script>
      // fetches app.js, which fetches test.js bundle 
      System.import('app');
    </script>
  </head>
</html>
```

```
// alternatively same use case as https://github.com/systemjs/systemjs/issues/1907
console.log(await System.import('test')); // empty object
console.log(await System.import('test')); // { default: 'hello world, useDefault: true }
```

From my limited viewpoint, it appears that either SystemJS should not allow this behavior (and throw an error) and force a dependency to already be loaded when it is requested, **or** change the resolution of bundled modules.

**Comment by khangiskhan** [comment]
Edit, i incorrectly referenced issue 1907, it should have been https://github.com/systemjs/systemjs/issues/1037

**Comment by joeldenning** [comment]
My understanding is that with SystemJS@>=2 each network request caused by `System.import()` represents exactly one module, whereas with SystemJS@<1 a single System.import() could register multiple modules.

My understanding for why systemjs@2 behaves that way is because that is what is specified in the [import maps spec](https://github.com/WICG/import-maps).

If you'd like to register multiple modules into systemjs with only one network request, that is possible. However, you cannot do it with `System.import()` and should instead do it with a regular script tag to your bundle. After that bundle is executed, you'll be able to import those modules or depend on them from other modules.

Another thing to note is that when you System.import() a bundle, you should not give that module a name. For example, you should do `define(['test'], function() {})` instead of `define('app', ['test'], function() {})`, since the named-register extra currently does not properly handle named exports during the instantiation of a named module.

Example (not tested):
```html
<!-- index.html -->
<html>
  <head>
    <script type="systemjs-importmap">
      {"imports": {"app": "/app.js"}}
    </script>
    <script src="https://cdn.jsdelivr.net/npm/systemjs@5.0.0/dist/system.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/systemjs@5.0.0/dist/extras/amd.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/systemjs@5.0.0/dist/extras/named-register.js">
</script>
    <script src="/bundled-deps.js"></script>
    <script>
      System.import('app')
    </script>
  </head>
</html>
```

```js
// app.js
define(['test'], function(test) {
  console.log(test) // I think this will work
});
```

```js
// bundled-deps.js
define('testDep', [], function(){});
define('test', ['testDep'] function(testDep) {
  var test = 'hello world';
  return test;
});

// Alternatively, you can call System.register, which doesn't need the AMD extra
System.register('testDep', [], function(_export) {
  return {}
})
System.register('test', ['testDep'], function(_export) {
  return {
    execute: function() {
      _export('default', 'hello world')
    }
  }
})

// Or you can call System.set() directly
System.set('test', {default: 'hello world'})
```

**Comment by guybedford** [comment]
This should be resolved in #9184.

**Comment by guybedford** [comment]
Released in 6.0.0.
